tensorboard/manager_test.py

```diff
@@ -254,17 +255,22 @@ def test_arguments_list_vs_tuple_irrelevant(self):
     self.assertEqual(with_list, with_tuple)
 
 
-class TensorBoardInfoIoTest(tf.test.TestCase):
+class TensorBoardInfoIoTest(unittest.TestCase):
   """Tests for `write_info_file`, `remove_info_file`, and `get_all`."""
 
   def setUp(self):
     super(TensorBoardInfoIoTest, self).setUp()
-    patcher = mock.patch.dict(os.environ, {"TMPDIR": self.get_temp_dir()})
-    patcher.start()
-    self.addCleanup(patcher.stop)
+    self.temp_dir = tempfile.mkdtemp()
+    self.patcher = mock.patch.dict(os.environ, {"TMPDIR": self.temp_dir})
+    self.patcher.start()
     tempfile.tempdir = None  # force `gettempdir` to reinitialize from env
     self.info_dir = manager._get_info_dir()  # ensure that directory exists
 
+  def tearDown(self):
```

Comment:
Can we just add more cleanups instead of using `tearDown`? As written,
`self.temp_dir` won’t be properly cleaned up if `self.patcher.stop()`
fails for any reason.